x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPlugin.java

```diff
@@ -138,16 +139,52 @@ public class EnrichPlugin extends Plugin implements SystemIndexPlugin, IngestPlu
         Setting.Property.NodeScope
     );
 
+    /**
+     * This setting solely exists because the original setting was accidentally renamed in
+     * https://github.com/elastic/elasticsearch/pull/111412.
+     */
+    @UpdateForV10(owner = UpdateForV10.Owner.DATA_MANAGEMENT)
```

Comment:
Are the `Settings.Property.Deprecated` below and this `@Update` too soon?

Reply:
I think the `@UpdateForV10` is fine, but the deprecated thing I'm not sure about yet. What I don't like about it here is that we can't redirect users back to the setting they _should_ be using (`cache_size`). Perhaps it'd be worth adding it into the constructor rather than the setting itself? Both is probably fine (the annotation would let us see it in the migration APIs I think)